### DISCUSSION TITLE
Understanding comoving and physical coordinates

### DISCUSSION THREAD

**Discussion by rizman18** [pfhopkins/gizmo-public#7] [General]
Hi, I have a question regarding the different between comoving and physical coordinates in FIRE-2, especially when I tried to read RJ simulation.

In the paper titled "Public Data Release of the FIRE-2 Cosmological Zoom-in Simulations of Galaxy Formation" it (subsection 4.2, number 3) is said that 
"Coordinates (h^{−1} kpc comoving): 3D positions; multiply by the scale factor a to convert to physical position." 
I think that means not the scale factor but the Hubble parameter. 

Using GIZMO to read snapshot 600 of Romeo-Juliet simulation (with h = 0.68 and of course, a = 1), GIZMO gives this information

host1 position = (64622.39, 64167.62, 66665.68) [kpc comoving] (hereafter xg, yg, zg) 
host2 position = (63850.66, 64103.79, 66342.77) [kpc comoving]

Using other python library, I got the host1 position in about xp = 44900, yp = 43600 (h^{-1} kpc comoving, based on the paper)
![image](https://github.com/user-attachments/assets/c82b71f1-3372-40b7-b408-ecb3985cbefa)

To synchronize the host1 position from GIZMO (in kpc comoving) and default position (in (h^{-1} kpc comoving), I can

- divide xp by h or  
- multiply xg by h so it will fulfill xg = xp / h (unit in kpc comoving)

But, what is the physical position? xg from GIZMO is in kpc comoving, xp is in h^{-1} kpc comoving. 
When I multiply xp by h it will give me kpc comoving, not kpc (based on the paper, the unit is h^{−1} kpc comoving)

Thanks

**Comment by mikegrudic** [comment]
Hi there,

In all cases, physical coordinates = a * comoving coordinates, where a is the cosmological scale factor.

The unit of either is kpc/h for FIRE simulation outputs, so if you want them in kpc you must divide by h.

Mike

**Reply by rizman18** [reply]
Thank you. Really appreciate it.
One more question.

Using halo_analysis, I got one of the halo (m = 1.23e10 Msun) near host1 (Romeo, index 600) lies on x1, y1, z1 = 9.324219, -79.054690, -57.367188 from hal['host.distance'] argument.
![image](https://github.com/user-attachments/assets/e4e71660-e104-4cc9-ad63-ecc0948a4d44)

After checking the density of dark particle using gizmo_analysis, I couldn't find that halo. Here is the density map
![image](https://github.com/user-attachments/assets/c9f54029-315e-4f2e-ba28-4a4fabcc9119)

Do  you have any suggestion? 

Thank you.
